src/pages/modals/form/formField/FormField.jsx

```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+import styles from './FormField.module.css';
+
+export const INPUT_TYPES = {
+  text: 'text',
+  date: 'date',
+};
+
+export class FormField extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: INPUT_TYPES.text,
+      style: null,
+    };
+  }
+
+  componentDidMount() {
+    const { type } = this.props;
+    if (type === INPUT_TYPES.date) {
+      this.setState({ style: INPUT_TYPES.date });
+    }
+  }
+
+  setType = (value) => {
+    const { type } = this.props;
+    if (type === INPUT_TYPES.date) {
+      this.setState({ type: value });
+    }
+  };
+
+  render() {
+    const { onChange, placeholder, value, fieldName, readOnly, required } = this.props;
+    const { type, style } = this.state;
+
+    return (
+      <div className={styles.formField}>
+        <div className={styles.fieldName}>{fieldName}</div>
+        <input
+          type={type}
+          required={required}
+          disabled={readOnly}
+          className={`${styles.input} ${styles[style]}`}
+          placeholder={placeholder}
+          onFocus={() => this.setType(INPUT_TYPES.date)}
+          value={value}
+          onChange={(event) => onChange(event.currentTarget.value)}
```

Comment:
create handler method, you can spread event inside handler

Reply:
```suggestion
export function FormField({ onChange, placeholder, inputValue, fieldName, readOnly, stylingType, inputName, error }) {
  const onChangeHandler = ({ target: { name, value } }) => {
    onChange(name, value);
  };

  return (
    <div className={styles.formField}>
      <div className={styles.fieldName}>{fieldName}</div>
      <div>
        <input
          type={stylingType}
          disabled={readOnly}
          name={inputName}
          className={styles.input}
          placeholder={placeholder}
          value={inputValue}
          onChange={onChangeHandler}
        />
        {error && <div className={styles.errorMessage}>{error}</div>}
      </div>
    </div>
  );
}
```
Rewrite this component